protocols/directfund/request.go

```diff
@@ -0,0 +1,40 @@
+package directfund
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/protocols/shared"
+)
+
+//go:generate msgp
+
+const DirectFundRequestMethod = "direct_fund"
+
+type DirectFundRequest struct {
+	CounterParty      string                   `msg:"counter_party"`
+	ChallengeDuration uint32                   `msg:"challenge_duration"`
+	Outcome           []shared.SingleAssetExit `msg:"outcome"`
+	AppDefinition     string                   `msg:"app_definition"`
+	AppData           []byte                   `msg:"app_data"`
+	Nonce             uint64                   `msg:"nonce"`
+}
```

Comment:
what is this encoding for?

Reply:
this is to serialize the request using msg pack